test/e2e/framework/aws/kinesis/kinesis.go

```diff
@@ -65,6 +65,37 @@ func PutRecord(kc kinesisiface.KinesisAPI, name string) string /*seqNumber*/ {
 	return *putOutput.SequenceNumber
 }
 
+// GetRecords get records from a kinesis data stream.
+func GetRecords(kc kinesisiface.KinesisAPI, name string) []*kinesis.Record {
+	shards, err := kc.ListShards(&kinesis.ListShardsInput{
+		StreamName: &name,
+	})
+	if err != nil {
+		framework.FailfWithOffset(2, "Failed to get shards from stream: %s", err)
+	}
+	if len(shards.Shards) == 0 {
+		framework.FailfWithOffset(2, "Shards not found from stream: %s", err)
+	}
+
+	shardIterator, err := kc.GetShardIterator(&kinesis.GetShardIteratorInput{
+		ShardId:           shards.Shards[0].ShardId,
```

Comment:
If the stream contains more than one shard, we're going to have to iterate over all of them: 

https://github.com/triggermesh/triggermesh/blob/92e6218a3772a4cc71653f5c1d0ede6d094b4671/pkg/sources/adapter/awsdynamodbsource/adapter.go#L212

(The example above is from the DynamoDB source, because DynamoDB streams are backed by Kinesis)